src/components/clusters/wizards/osd/CreateOsdWizardFooter.test.tsx

```diff
@@ -63,4 +65,30 @@ describe('<CreateOsdWizardFooter />', () => {
     render(<CreateOsdWizardFooter {...props} />);
     expect(screen.getByTestId(wizardPrimaryBtnTestId)).not.toHaveAttribute('aria-disabled');
   });
+
+  it("Doesn't proceed to the next step when validation fails", () => {
+    mockedUseFormState.mockReturnValue({
+      ...useFormStateReturnValue,
+      validateForm: jest.fn().mockResolvedValue({ mockError: 'Mock Error' }),
+    });
+
+    const { user } = render(<CreateOsdWizardFooter {...props} />);
+    const nextButton = screen.getByTestId(wizardPrimaryBtnTestId);
+
+    user.click(nextButton);
```

Comment:
the user click action should be `await`ed on!

otherwise the following assertion might yield false results (a false-negative in this case).

Reply:
(and same comment for other similar instances of un`await`ed user-clicks)